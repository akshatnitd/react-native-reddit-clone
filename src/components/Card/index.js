import React from "react"
import PropTypes from "prop-types"
import { View, StyleSheet } from "react-native"
import { colorStyle, fontStyle } from "../../style"

const Card = (props) => {
  const { containerStyle, children } = props
  return (
    <View style={[style.container, containerStyle]} {...props}>
      {children}
    </View>
  )
}

Card.propTypes = {
  containerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  children: PropTypes.node,
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    flexWrap: "nowrap",
    backgroundColor: colorStyle.superLight,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 1.5,
    elevation: 1,
  },
})

export default Card
