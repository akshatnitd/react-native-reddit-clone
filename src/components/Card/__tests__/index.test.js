import React from "react"
import renderer from "react-test-renderer"

import Card from ".."

it("renders correctly", () => {
  const tree = renderer.create(<Card />)
  expect(tree.toJSON()).toMatchSnapshot()
})
