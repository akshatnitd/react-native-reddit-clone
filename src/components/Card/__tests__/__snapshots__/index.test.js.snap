// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f2f2f2",
        "elevation": 1,
        "flex": 1,
        "flexWrap": "nowrap",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      undefined,
    ]
  }
/>
`;
